Naming/SmartUniqueNamer.php

```diff
@@ -17,10 +17,45 @@ public function name($object, PropertyMapping $mapping): string
     {
         $file = $mapping->getFile($object);
         $originalName = $file->getClientOriginalName();
-        $originalExtension = \pathinfo($originalName, PATHINFO_EXTENSION);
+        $originalExtension = \strtolower(\pathinfo($originalName, PATHINFO_EXTENSION));
         $originalBasename = \basename($originalName, '.'.$originalExtension);
         $originalBasename = Transliterator::transliterate($originalBasename);
+        $uniqId = \str_replace('.', '', \uniqid('-', true));
+        $uniqBasename = \sprintf('%s%s', $originalBasename, $uniqId);
+        $uniqExtension = \sprintf('%s.%s', $uniqId, $originalExtension);
```

Comment:
What's the point of concatenating uniqid to both basename **and** extension? 🤔

Reply:
On lines [44](https://github.com/aewens/VichUploaderBundle/blob/f0682b779f75e8da250699ac4e79cb0253f4c272/Naming/SmartUniqueNamer.php#L44), [47](https://github.com/aewens/VichUploaderBundle/blob/f0682b779f75e8da250699ac4e79cb0253f4c272/Naming/SmartUniqueNamer.php#L47), and [50](https://github.com/aewens/VichUploaderBundle/blob/f0682b779f75e8da250699ac4e79cb0253f4c272/Naming/SmartUniqueNamer.php#L50) the variable is used in the edge case where a file has an abnormally long extension; so the extension would be shortened instead of the basename. The variable helps reduce code duplication during that check, but I can move it down to that portion of the code to make it easier to see why it is being used.

Reply:
This has been fixed in commit [3cecc4c](https://github.com/aewens/VichUploaderBundle/commit/3cecc4c3e4c0ef831872a1cf097fc6be1ed605fa#diff-43549d537b7d4f88517af8b366fd4a62)